src/vault/client/ext/aws.clj

```diff
@@ -92,11 +96,12 @@
 
 (defmethod client/authenticate* :aws-iam
   [client _ aws-ctx]
-  (let [{:keys [iam-role credentials]} aws-ctx
+  (let [{:keys [iam-role credentials sts-region]} aws-ctx
+        signing-region (or sts-region "us-east-1")
```

Comment:
It would be safer to have this be `(if-not (str/blank? sts-region) sts-region "us-east-1")`.